code/game/machinery/alarm.dm

```diff
@@ -859,22 +859,22 @@ FIRE ALARM
 	return overlays_cache[key]
 
 /obj/machinery/firealarm/on_update_icon()
-	overlays.Cut()
+	cut_overlays()
 	icon_state = "casing"
 	if(construct_state && !istype(construct_state, /decl/machine_construction/wall_frame/panel_closed))
-		overlays += get_cached_overlay(construct_state.type)
+		add_overlay(get_cached_overlay(construct_state.type))
```

Comment:
As noted on Discord, `get_cached_overlay()` here seems redundant — just use the native istate cache in SSoverlays.